Framework/PythonInterface/plugins/algorithms/PeakMatching.py

```diff
@@ -0,0 +1,271 @@
+        with open(path, 'r') as file_to_read:
+            peak_data = json.load(file_to_read)
+        return peak_data
+
+    def PyExec(self):
+        table = self.getPropertyValue("PeakTable")
+        path = self.getPropertyValue("PeakDatabase")
+        peak_data = self.process_peak_data(path)
+
+        sigma_column = self.getProperty("SigmaColumn").value
+        peak_column = self.getProperty("PeakCentreColumn").value
+        input_peaks = self.get_input_peaks(table,peak_column,sigma_column)
+
+        all_data , primary_data, secondary_data = self.get_matches(peak_data, input_peaks)
+
+        output_table_names = [self.getPropertyValue("PrimaryPeaks"),
+                              self.getPropertyValue("SecondaryPeaks"),
+                              self.getPropertyValue("AllPeaks"),
+                              self.getPropertyValue("SortedByEnergy"),
+                              self.getPropertyValue("ElementLikelihood")]
+
+        self.output_data(primary_data, secondary_data, all_data,output_table_names)
+
+    def process_peak_data(self,path):
+        '''
+        opening the json file containing the peak data
+        Format of file:
+            {
+            'Element': {
+                'Z': z,
+                'A': a,
+                'Primary' : {
+                    'Transition':energy,
+                    ..., ...
+                    },
+                'Secondary': {
+                    'Transition': energy,
+                    ..., ...
+                    },
+                'Gammas': {
+                    'Isotope': energy,
+                    ..., ...
+                    },
+                'Electron': {
+                    'Energy': probability,
+                    ..., ...
+                    }
+                '''
+        peak_data = None
+        if path:
+            with open(path, 'r') as file_to_read:
+                peak_data = json.load(file_to_read)
+        else:
+            peak_data = self.get_default_peak_data()
+
+        processed_data = {}
+        processed_data["All energies"] = {}
+        for label in LABELS:
+            processed_data[LABELS[label]] = {}
+
+        for element in peak_data:
+            processed_data["All energies"][element] = {}
+            for label in LABELS:
+                energy_list = [float(energy) for transition,energy in list(peak_data[element][label].items())]
+                transition_list = [transition for transition,energy in list(peak_data[element][label].items())]
+                processed_data[LABELS[label]][element] = dict(zip(transition_list, energy_list))
+                processed_data["All energies"][element].update(dict(zip(transition_list, energy_list)) )
+
+        return processed_data
+
+    def make_count_table(self,name,data):
+        table = CreateEmptyTableWorkspace(OutputWorkspace=name)
+
+        table.addColumn("str", "Element")
+        table.addColumn("int", "Likelihood(au)")
```

Comment:
why is the likelihood measured in atomic units? Are you sure that is the best word to use here?

Reply:
I meant arbitary units

Reply:
its normal to put it in full or if you need to have an abbreviation "A.U."